pkg/cloudprovider/plugins/aws/aws_ec2.go

```diff
@@ -248,77 +216,86 @@ func (ec2Cfg *ec2ServiceConfig) getInstances() ([]*ec2.Instance, error) {
 			MaxResults: aws.Int64(internal.MaxCloudResourceResponse),
 			Filters:    filter,
 		}
-		filterInstances, e := ec2Cfg.apiClient.pagedDescribeInstancesWrapper(request)
-		if e != nil {
-			return nil, e
+		filterInstances, err := ec2Cfg.apiClient.pagedDescribeInstancesWrapper(request)
+		if err != nil {
+			return nil, err
 		}
 		instances = append(instances, filterInstances...)
 	}
-
-	awsPluginLogger().V(1).Info("Vm instances from cloud", "account", ec2Cfg.accountNamespacedName,
-		"instances", len(instances))
+	awsPluginLogger().Info("Vm instances from cloud", "account", ec2Cfg.accountNamespacedName,
+		"selector", namespacedName, "instances", len(instances))
 
 	return instances, nil
 }
 
-// DoResourceInventory gets inventory from cloud for given cloud account.
+// DoResourceInventory gets inventory from cloud for a given cloud account.
 func (ec2Cfg *ec2ServiceConfig) DoResourceInventory() error {
 	vpcs, err := ec2Cfg.getVpcs()
 	if err != nil {
 		awsPluginLogger().Error(err, "failed to fetch cloud resources", "account", ec2Cfg.accountNamespacedName)
 		return err
 	}
+	vpcNameToID := ec2Cfg.buildMapVpcNameToID(vpcs)
+	vpcPeers, _ := ec2Cfg.buildMapVpcPeers()
 
-	instances, err := ec2Cfg.getInstances()
-	if err != nil {
-		awsPluginLogger().Error(err, "failed to fetch cloud resources", "account", ec2Cfg.accountNamespacedName)
-		return err
-	} else {
-		exists := struct{}{}
-		vpcIDs := make(map[string]struct{})
-		instanceIDs := make(map[internal.InstanceID]*ec2.Instance)
-		vpcNameToID := ec2Cfg.buildMapVpcNameToID(vpcs)
-		vpcPeers, _ := ec2Cfg.buildMapVpcPeers()
+	// Call cloud APIs for the configured CloudEntitySelectors CRs.
+	if len(ec2Cfg.selectors) == 0 {
+		awsPluginLogger().V(1).Info("Fetching vm resources from cloud skipped",
+			"account", ec2Cfg.accountNamespacedName, "resource-filters", "not-configured")
+		vmSnapshot := make(map[types.NamespacedName][]*ec2.Instance)
```

Comment:
Do we need to create a map everytime when there are no selectors configured?
can't we just initialize it as nil?